components/camel-sql/src/main/java/org/apache/camel/component/sql/stored/SqlStoredProducer.java

```diff
@@ -123,8 +123,8 @@ private StatementWrapper createStatement(Exchange exchange) throws SQLException
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```

Comment:
This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs